include/deal.II/meshworker/scratch_data.h

```diff
@@ -0,0 +1,1301 @@
+     * @param face_quadrature Face quadrature, used for FEFaceFvalues and
+     * FESubfaceValues for both the current cell and the neighbor cell
+     * @param face_update_flags UpdateFlags used for FEFaceFvalues and
+     * FESubfaceValues for the current cell
+     * @param neighbor_face_update_flags UpdateFlags used for FEFaceFvalues and
+     * FESubfaceValues for the neighbor cell
+     */
+    ScratchData(
+      const FiniteElement<dim, spacedim> &fe,
+      const Quadrature<dim> &             quadrature,
+      const UpdateFlags &                 update_flags,
+      const Quadrature<dim - 1> &face_quadrature   = Quadrature<dim - 1>(),
+      const UpdateFlags &        face_update_flags = update_default);
+
+    /**
+     * Same as the other constructor, using the default MappingQ1.
+     *
+     * @param mapping The mapping to use in the internal FEValues objects
+     * @param fe The finite element
+     * @param quadrature The cell quadrature
+     * @param update_flags UpdateFlags for the current cell FEValues
+     * @param neighbor_update_flags UpdateFlags for the neighbor cell FEValues
+     * @param face_quadrature Face quadrature, used for FEFaceFvalues and
+     * FESubfaceValues for both the current cell and the neighbor cell
+     * @param face_update_flags UpdateFlags used for FEFaceFvalues and
+     * FESubfaceValues for the current cell
+     * @param neighbor_face_update_flags UpdateFlags used for FEFaceFvalues and
+     * FESubfaceValues for the neighbor cell
+     */
+    ScratchData(
+      const FiniteElement<dim, spacedim> &fe,
+      const Quadrature<dim> &             quadrature,
+      const UpdateFlags &                 update_flags,
+      const UpdateFlags &                 neighbor_update_flags,
+      const Quadrature<dim - 1> &face_quadrature   = Quadrature<dim - 1>(),
+      const UpdateFlags &        face_update_flags = update_default,
+      const UpdateFlags &        neighbor_face_update_flags = update_default);
+
+    /**
+     * Deep copy constructor. FEValues objects are not copied.
+     */
+    ScratchData(const ScratchData<dim, spacedim> &scratch);
+
+    /**
+     * Initialize the internal FEValues with the given @p cell.
+     */
+    const FEValues<dim, spacedim> &
+    reinit(
+      const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell);
+
+    /**
+     * Initialize the internal FEFaceValues to use the given @p face_no on the given
+     * @p cell.
+     */
+    const FEFaceValues<dim, spacedim> &
+    reinit(const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+           const unsigned int face_no);
+
+    /**
+     * Initialize the internal FESubfaceValues to use the given @p subface_no,
+     * on @p face_no, on the given @p cell.
+     *
+     * If @p subface_no is numbers::invalid_unsigned_int, the reinit() function
+     * that takes only the @p cell and the @p face_no is called.
+     */
```

Comment:
here and in the following functions, can you state the effect on what `get_current_fe_values()` will return after this function is called?